singularity-eos/eos/eos_multi_eos.hpp

```diff
@@ -699,17 +704,17 @@ class MultiEOS : public EosBase<MultiEOS<BulkModAvgT, GruneisenAvgT, EOSModelsT.
     density_tot = 0;
     Real spvol_tot = 0;
     for (size_t m = 0; m < nmat_; m++) {
-      spvol_tot += robust::ratio(mass_fracs, density_mat[m]);
-      sie_tot += robust::ratio(mass_fracs, sie_mat[m]);
+      spvol_tot += robust::ratio(mass_fracs[m], density_mat[m]);
+      sie_tot += robust::ratio(mass_fracs[m], sie_mat[m]);
     }
     PORTABLE_REQUIRE(spvol_tot > 0., "Material volumes sum to zero");
     density_tot = robust::ratio(1.0, spvol_tot);
   }
 
   PORTABLE_FORCEINLINE_FUNCTION
-  auto CreateEOSArray() {
+  auto CreateEOSArray() const {
```

Comment:
There's an "experimental" C++ feature [`make_array`](https://en.cppreference.com/w/cpp/experimental/make_array.html) that you may consider here, with a provided implementation. Not really "better" but could be a utility useful in other parts of the code.

Reply:
I'm not sure I understand the `VT` type logic explained there, but it seems like I'd need to cast each of these models to the variant type before sending the models to `make_array`, right? Can this be efficiently done for an argument pack?

Reply:
`VT` is just value type and wouldn't be used at the call site. The usage could be just a drop in to your code: 

```c++
[this](auto &&...eos_models) { return make_array(eos_models...); },
        models_);
```
which, now that I write it, probably is a little overkill for what we need.

Reply:
`VT` is deduced through [`std::common_type`](https://en.cppreference.com/w/cpp/types/common_type.html) (if not specified explicitly), so the array value type is deduced through the arguments.